home/migrations/0099_migrate_content_page_templates_to_standalone_templates.py

```diff
@@ -1,37 +1,105 @@
+        .filter(count__gt=1)
+        .values_list("whatsapp_title", flat=True)
+    )
+    for whatsapp_title in duplicate_whatsapp_titles:
+        pages = ContentPage.objects.filter(whatsapp_title=whatsapp_title)
+        while pages.count() > 1:
+            page = pages.first()
+            suffix = 1
+            candidate_whatsapp_title = whatsapp_title
+            while ContentPage.objects.filter(
+                whatsapp_title=candidate_whatsapp_title
+            ).exists():
+                suffix += 1
+                candidate_whatsapp_title = f"{whatsapp_title}-{suffix}"
+            page.whatsapp_title = candidate_whatsapp_title
+            page.save(update_fields=["whatsapp_title"])
+
+
+def rename_matching_whatsapp_names(ContentPage: Any, WhatsAppTemplate: Any) -> None:
+    templates = WhatsAppTemplate.objects.all()
+    for template in templates:
+        pages = ContentPage.objects.filter(whatsapp_title=template.name)
+        while pages.count() > 1:
+            page = pages.first()
+            suffix = 1
+            candidate_whatsapp_title = template.name
+            while ContentPage.objects.filter(
+                whatsapp_title=candidate_whatsapp_title
+            ).exists():
+                suffix += 1
+                candidate_whatsapp_title = f"{template.name}-{suffix}"
+            page.whatsapp_title = candidate_whatsapp_title
+            page.save(update_fields=["whatsapp_title"])
 
 
 def migrate_content_page_templates_to_standalone_templates(
-    ContentPage: Any, WhatsAppTemplate: Any
+    ContentPage: Any, WhatsAppTemplate: Any, Image: Any
 ) -> None:
     content_pages = ContentPage.objects.filter(is_whatsapp_template=True)
 
     for content_page in content_pages:
         whatsapp_block = content_page.whatsapp_body[0]
         whatsapp_value = whatsapp_block.value
-        whatsapp_template = WhatsAppTemplate.objects.create(
-            name=content_page.whatsapp_template_name,
-            locale=content_page.locale,
-            message=whatsapp_value.get("message", ""),
-            example_values=whatsapp_value.get("example_values", []),
-            category=content_page.whatsapp_template_category,
-            buttons=whatsapp_value.get("buttons", []),
-            image=whatsapp_value.get("image", None),
-            submission_status="NOT_SUBMITTED_YET",
-            submission_result="",
-        )
-        content_page.whatsapp_body = []
-        content_page.whatsapp_body.append(("Whatsapp_Template", whatsapp_template))
-        content_page.is_whatsapp_template = False
-        content_page.save()
+        print(f"Content Page slug: {content_page.slug}")
+        print(f"Content page title: {content_page.whatsapp_title}")
+        print(f"Whatsapp Value type: {type(whatsapp_value)}, value: {whatsapp_value}")
+        if (
+            not hasattr(whatsapp_value, "_meta")
+            or whatsapp_value._meta.verbose_name != "WhatsApp Template"
+        ):
```

Comment:
Could we not rather do the inverse of this, and use the `continue` keyword, so that we can avoid this deep nesting? e.g.
```python
if hasattr(whatsapp_value, "_meta") and whatsapp_value._meta.verbose_name == "WhatsApp Template":
    continue
```